lib/coverage.js

```diff
@@ -390,18 +391,19 @@ exports.analyze = function (options) {
     return cov;
 };
 
-internals.addSourceMapsInformation = function (ret, num) {
+internals.addSourceMapsInformation = function (smc, ret, num) {
 
     const position = {
         source: ret.filename,
         line: num,
         column: 0
     };
-    const originalPosition = SourceMapSupport.mapSourcePosition(position);
+    let originalPosition = smc.originalPositionFor(position);
     const source = ret.source[num];
 
     if (position !== originalPosition) {
-        source.originalFilename = originalPosition.source.replace(Path.join(process.cwd(), '/').replace(/\\/g, '/'), '');
+        // Ensure folder separator to be url-friendly
+        source.originalFilename = originalPosition.source && originalPosition.source.replace(/\\/g, '/');
```

Comment:
Sourcemaps always expect path to use `/` separator, as URLs does, so we need to do the replacement.
Removing process.cwd() is useless as the sourcemap won't contain it.